ReactNativeClient/lib/models/Setting.js

```diff
@@ -435,6 +435,31 @@ class Setting extends BaseModel {
 			'style.sidebar.width': { value: 150, minimum: 80, maximum: 400, type: Setting.TYPE_INT, public: false, appTypes: ['desktop'] },
 			'style.noteList.width': { value: 150, minimum: 80, maximum: 400, type: Setting.TYPE_INT, public: false, appTypes: ['desktop'] },
 
+			'layout.folderList.factor': {
+				value: 1,
+				type: Setting.TYPE_INT,
+				section: 'appearance',
+				public: true,
+				appTypes: ['cli'],
+				label: () => _('Factor of screen Folder List covers'),
```

Comment:
"Factor of screen Folder List covers". If you ask around you and say that to someone, will they understand what you mean? Also as tessus said, a number without a unit is meaningless.

Reply:
I'm not sure how to best explain the user how this factor works. Maybe "Folder list growth factor"?

Reply:
would "ratio" a better expression?

Reply:
Ratio sounds good! Although I could make it more user understandable by exposing percentages, and processing the percentages for the actual factors in `app-gui.js`

So, I could enforce that `layout.FolderList.factor` + `layout.NoteList.factor` + `layout.note.factor` = 100, and calculate factors by 1 / `layout.FolderList.factor` for example. This would also take care of the issue of having numbers without a unit.

Reply:
It's not a percentage though. If you have three components, one with 100%, another with 100% and the last one with 300%, what would it mean? They take 500% of the available space?

Let's keep it an integer and try to convey the meaning as clearly as possible. Perhaps we need to use the "description" field to provide a detailed explanation. The concept comes from the CSS "flex" property so maybe we can copy their doc.

"1" is the base unit. A factor of "2" would mean that the component takes twice as much space a component with factor "1", etc.

Reply:
> It's not a percentage though

Well, it's not, but it just seems like a intuitive way to convey what we want to say. My suggestion was also to enforce the summation of the three values to be 100%. So, 100%, 100%, and 300% would throw an error (even though it really isn't an error) so as to make it more user friendly.

> The concept comes from the CSS "flex" property so maybe we can copy their doc.

That sounds very good! I'll do that. From https://developer.mozilla.org/en-US/docs/Web/CSS/flex, "The flex CSS property sets how a flex item will grow or shrink to fit the space available in its flex container". So we could make it `The Folder List Factor setting sets how the folder list will grow or shrink to fit the available space in its container with respect to the other containers.`

Reply:
Or possibly `The Folder List Factor setting sets how the folder list component will grow or shrink based on the factors of the other components.`

Reply:
> > The concept comes from the CSS "flex" property so maybe we can copy their doc.
> 
> That sounds very good! I'll do that. From https://developer.mozilla.org/en-US/docs/Web/CSS/flex, "The flex CSS property sets how a flex item will grow or shrink to fit the space available in its flex container". So we could make it `The Folder List Factor setting sets how the folder list will grow or shrink to fit the available space in its container with respect to the other containers.`

I like this

Reply:
Maybe integer is not the best idea. This means I can make one element only twice as big as the other. What If I wanted it 1.5x wider?

Reply:
Hmm that seems right @tessus. I don't know a better solution though. Any suggestions?

Reply:
use integer as percentage 
100 = 1x
150 = 1.5x
200 = 2x

Reply:
Yes, as PackElend said, the user can use 100, 150, etc. to express whatever ratio they need. We don't need to multiply the number ourselves or make it float.

Reply:
For the description, please use this for the three properties:

> The factor property sets how the item will grow or shrink to fit the available space in its container with respect to the other items. Thus an item with a factor of "2" will take twice as much space as an item with a factor of "1".

Reply:
Cool!

Since this message is so large, I've only added the description for the last factor in order to avoid redundancy. Please let me know if that's a good decision, and if I should revert that.